connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetrics.java

```diff
@@ -0,0 +1,220 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.runtime;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.JmxReporter;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.connect.runtime.distributed.WorkerGroupMember;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * The Connect metrics with JMX reporter.
+ */
+public class ConnectMetrics {
+
+    public static final String JMX_PREFIX = "kafka.connect";
+    public static final String WORKER_ID_TAG_NAME = "worker-id";
+    private static final AtomicInteger CONNECT_WORKER_ID_SEQUENCE = new AtomicInteger(1);
+
+    /**
+     * Utility to ensure the supplied name contains valid characters, replacing sequences of 1 or more
+     * ':', '/', '\', '*', '?', '[', ']', '=', or comma characters with a single '-'.
+     *
+     * @param name the name; may not be null
+     * @return the validated name; never null
+     */
+    static String makeValidName(String name) {
+        Objects.requireNonNull(name);
+        name = name.trim();
+        if (!name.isEmpty()) {
+            name = name.replaceAll("[:/\\\\*?,;\\[\\]\\\\=]+", "-");
+        }
+        return name;
+    }
+
+    private final Logger log;
+    private final Metrics metrics;
+    private final Time time;
+    private final String workerId;
+    private final Map<String, MetricGroup> groupsByName = new HashMap<>();
+
+    public ConnectMetrics(String workerId, WorkerConfig config, Time time) {
+        this.log = LoggerFactory.getLogger(WorkerGroupMember.class);
+        this.workerId = workerId != null ? makeValidName(workerId) : "worker-" + CONNECT_WORKER_ID_SEQUENCE.getAndIncrement();
+        this.time = time;
+
+        Map<String, String> metricsTags = new LinkedHashMap<>(); // currently empty
```

Comment:
Is this a placeholder for future use? `MetricsConfig` is initialized with an empty `LinkedHashMap` within its constructor?

Reply:
Well, sort of. It took me a long time to figure out how to pass tags, and I started out passing some global tags that I've since removed. Happy to remove it.